src/api/haventec.authenticate.ts

```diff
@@ -0,0 +1,65 @@
+import { HT_DataService } from "../helpers/ht_data.service";
+import { HT_Error, ErrorCode, ErrorMessage } from "../model/error";
+import { ISessionUpdateRequestObject } from "../model/session.update.request.object";
+
+export class HaventecAuthenticate {
+
+    private ht_dataService: HT_DataService;
+
+    constructor(username: string) {
+        if (!username) throw  new HT_Error(ErrorCode.HT_AN_PARAM_ERROR, ErrorMessage.INSUFFICIENT_PARAMETERS);
+        this.ht_dataService = new HT_DataService(username.toLowerCase());
```

Comment:
we have the .toLowerCase() already at the HT_DataService, so we could delete it from here. The DataService will take care of initialise it properly